### ISSUE TITLE
[B60][torch-xpu-ops][XPU] STFT complex128 gradient tests hit UR_RESULT_ERROR_DEVICE_LOST

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#4796]
## Summary

B60 shows `UR_RESULT_ERROR_DEVICE_LOST` in torch-xpu-ops STFT complex128 gradient tests. The same tests passed on PVC.

## Failure Category

Device lost during STFT complex128 gradient/autograd execution.

Representative message:

```text
UR_RESULT_ERROR_DEVICE_LOST
```

## Scope

- Test suite: torch-xpu-ops
- Platform comparison: B60 error/failed; PVC passed
- Affected rows observed: 3

## Failures

Direct torch-xpu-ops XPU test files:

```text
test/xpu/test_ops_gradients_xpu.py::TestBwdGradientsXPU::test_fn_grad_stft_xpu_complex128
test/xpu/test_ops_fwd_gradients_xpu.py::TestFwdGradientsXPU::test_fn_fwgrad_bwgrad_stft_xpu_complex128
test/xpu/test_ops_gradients_xpu.py::TestBwdGradientsXPU::test_fn_gradgrad_stft_xpu_complex128
```

All fail with:

```text
UR_RESULT_ERROR_DEVICE_LOST
```

## Expected Behavior

The STFT complex128 gradient paths should complete without device loss, consistent with PVC behavior.

## Notes

All visible rows are STFT + complex128 gradient variants, so this appears to be a focused operator/autograd failure family.

## Full Case List

Total cases: 3

1. `torch-xpu-ops` | `test/xpu/test_ops_gradients_xpu.py::TestBwdGradientsXPU::test_fn_grad_stft_xpu_complex128` | `error`
   - `UR_RESULT_ERROR_DEVICE_LOST`
2. `torch-xpu-ops` | `test/xpu/test_ops_fwd_gradients_xpu.py::TestFwdGradientsXPU::test_fn_fwgrad_bwgrad_stft_xpu_complex128` | `failed`
   - `UR_RESULT_ERROR_DEVICE_LOST`
3. `torch-xpu-ops` | `test/xpu/test_ops_gradients_xpu.py::TestBwdGradientsXPU::test_fn_gradgrad_stft_xpu_complex128` | `failed`
   - `UR_RESULT_ERROR_DEVICE_LOST`

## Reproducer / Environment Setup

```bash
docker run -it -e TZ=Asia/Shanghai --device=/dev/mem --device=/dev/dri --group-add video --privileged -v $(realpath ${HOME}):/home/jenkins --shm-size=8g intelgpu/ubuntu-26.04-rolling:26.18 bash

apt update
apt upgrade -y

curl -LsSf https://astral.sh/uv/install.sh | sh
source $HOME/.local/bin/env
uv venv myvenv --python 3.10 --clear
source myvenv/bin/activate
uv pip install pip wheel "setuptools<81"
uv pip install torch==2.13.0 torchaudio torchvision --index-url https://download.pytorch.org/whl/xpu

pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout ${pytorch_commit}

uv pip install -r .ci/docker/requirements-ci.txt
uv pip install -U typing_extensions
uv pip install pytest pytest-timeout pytest-xdist pytest-rerunfailures

cd third_party/
git clone https://github.com/intel/torch-xpu-ops
xpu_ops_commit="$(cat xpu.txt)"
cd torch-xpu-ops
git checkout $xpu_ops_commit
cd test/xpu

python -m pytest test_ops_gradients_xpu.py -k 'test_fn_grad_stft_xpu_complex128 or test_fn_gradgrad_stft_xpu_complex128'
python -m pytest test_ops_fwd_gradients_xpu.py -k 'test_fn_fwgrad_bwgrad_stft_xpu_complex128'
```

**Comment by ZhaoqiongZ** [comment]
@torchxpubot fix

**Comment by torchxpubot** [comment]
## Issue Triage

| Field | Value |
|-------|-------|
| Reproduction missing | no |
| Scope | torch-xpu-ops |
| Dependencies | (none) |
| Handling | agent-fixable |

*Automated by issue-triage.*



## Reproduce (stage=nightly)

**Verdict: `NOT_REPRODUCED`** — all three reported cases pass, and pass fast.

Environment used:

| | |
|---|---|
| GPU | 8x `Intel(R) Arc(TM) Pro B60 Graphics` (BMG), driver `1.15.38646+7` |
| torch | `2.15.0.dev20260905+xpu` (pytorch `11898ed9`) |
| torch-xpu-ops | `09ace278` (main), test files from this tree |
| Env | `PYTORCH_TEST_WITH_SLOW=1`, run from `third_party/torch-xpu-ops/test/xpu` |

Each node id was run **individually** under a hard 660s cap on cards 0-3, matching
the reproducer in #4947:

| Case | Result (cards 0/1/2/3) |
|---|---|
| `test_ops_gradients_xpu.py::TestBwdGradientsXPU::test_fn_grad_stft_xpu_complex128` | PASSED, ~7.7-8.0s |
| `test_ops_gradients_xpu.py::TestBwdGradientsXPU::test_fn_gradgrad_stft_xpu_complex128` | PASSED, ~9.4-9.6s |
| `test_ops_fwd_gradients_xpu.py::TestFwdGradientsXPU::test_fn_fwgrad_bwgrad_stft_xpu_complex128` | PASSED, ~8.2-8.5s |

12/12 individual runs passed. The group was also run 3x with all 8 cards visible
(no `ZE_AFFINITY_MASK`): 11 passed / 12 skipped / 1 xfailed each time, ~26-32s
total. No `UR_RESULT_ERROR_DEVICE_LOST`, no hang, no worker crash.

The three `float64` siblings of the same family were checked as an extra health
datapoint and also pass in ~8s each, including
`test_fn_fwgrad_bwgrad_stft_xpu_float64` (tracked in #4996).



## Summary

- **Issue:** #4796 — [B60][torch-xpu-ops][XPU] STFT complex128 gradient tests hit `UR_RESULT_ERROR_DEVICE_LOST`
- **Path:** single-bug
- **Outcome:** `SKIPPED(reason=no_longer_reproduces)`
- **Fix produced:** none. The pipeline stopped at Stage 2, so no root-cause,
  no code change, no branch, and no patch artifact.

### Why this is a stale report rather than a live bug

The reported symptom is the same failure family as #4947, which describes it
precisely: these cases *hang past 10 minutes* on BMG, the pytest-xdist worker is
killed, and the collateral surfaces as `UR_RESULT_ERROR_DEVICE_LOST` /
`worker 'gwN' crashed`. On the current nightly and driver the same cases finish
in **8-10 seconds**, roughly two orders of magnitude short of the hang
threshold, consistently across four separate B60 cards. There is no slow path
left to fix here.

The original report was filed against `torch==2.13.0`; the double-precision
`stft` gradient paths behave normally on `2.15.0.dev20260905+xpu`.

### Follow-up for a human (deliberately not done here)

`test/xpu/skip_list_common.py` still carries skip entries for this family:

- `test_ops_gradients_xpu.py` -> `test_fn_gradgrad_stft_xpu_complex128` (cites #4947)
- `test_ops_fwd_gradients_xpu.py` -> `test_fn_fwgrad_bwgrad_stft_xpu_complex128` (cites #4947)
- `test_ops_fwd_gradients_xpu.py` -> `test_fn_fwgrad_bwgrad_stft_xpu_float64` (cites #4996)

All three now pass, so those decorators look stale. Removing a skip is its own
code change needing its own verification and PR, so this run did not touch them —
it is flagged as a candidate cleanup, ideally handled together with #4947 rather
than under this issue. Note `test_fn_grad_stft_xpu_complex128` (the case this
issue reports as `error`) was never skip-listed and passes today.

*Automated by issue-handler.*



_Generated by [fix job](https://github.com/intel/torch-xpu-ops/actions/runs/34093322093)._

**Comment by ZhaoqiongZ** [comment]
**weak-rejected** 

notes: the issue is not reproduced by bot because bot use OMIX  with specific driver version, so this should be a driver issue, thus wrong conclusion, need human to track this issue.
